utils/convert.mjs

```diff
@@ -512,51 +545,166 @@ function cleanUpClassItems(data) {
     }
   }
 
-  // Reduce the amount of information in each function's overloads, while
-  // keeping all the essential data available.
-  const flattenOverloads = funcObj => {
+  // Recursively processes a parameter object to normalize its structure.
+  // Handles function types, rest parameters, union types, array tuples, and nested properties.
+  const processParam = param => {
     const result = {};
 
-    const processOverload = overload => {
-      if (overload.params) {
-        return Object.values(overload.params).map(param => processParam(param));
+    if (!param.type) return result;
+
+    // Handle function parameters
+    if (param.type === 'Function') {
+      result.type = 'Function';
+
+      if (param.parameters) {
+        // Convert numeric-keyed object or array to array of processed params
+        if (Array.isArray(param.parameters)) {
+          result.parameters = param.parameters.map(processParam);
+        } else if (typeof param.parameters === 'object') {
+          result.parameters = Object.values(param.parameters).map(processParam);
+        }
       }
-      return overload;
+
+      if (param.optional) result.optional = true;
+      if (param.rest) result.rest = true;
+      return result;
     }
 
-    // To simplify `parameterData.json`, instead of having a separate field for
-    // optional parameters, we'll add a ? to the end of parameter type to
-    // indicate that it's optional.
-    const processParam = param => {
-      let type = param.type;
-      if (param.optional) {
-        type += '?';
+    // Handle rest parameters (e.g., "...string[]")
+    let typeStr = param.type;
+    if (typeStr.startsWith('...')) {
+      result.rest = true;
+      typeStr = typeStr.slice(3);
+      if (typeStr.endsWith('[]')) {
+        typeStr = typeStr.slice(0, -2);
       }
-      if (param.rest) {
-        type = `...${type}[]`;
+    }
+
+    // Parse union type strings
+    const parseUnion = unionStr => {
```

Comment:
Rather than parsing out arrays, tuples, etc here, could we modify `typeObject` to return something structured? Since we start with the structure, rather than condensing it and re-parsing it, we could just keep the original structure.